src/spark_rapids_ml/knn.py

```diff
@@ -90,15 +90,42 @@ class _NearestNeighborsCumlParams(_CumlParams, HasInputCol, HasLabelCol):
         typeConverter=TypeConverters.toInt,
     )
 
+    id_col = Param(
+        Params._dummy(),
+        "id_col",
+        "id column name.",
+        typeConverter=TypeConverters.toString,
+    )
+
     def setInputCol(self, value: str) -> "_NearestNeighborsCumlParams":
         """
         Sets the value of `inputCol`.
         """
         self.set_params(inputCol=value)
         return self
 
+    def setIdCol(self, value: str) -> "_NearestNeighborsCumlParams":
+        """
+        Sets the value of `id_col`. If not set, zipWithIndex will be used and the row order is not guaranted to be deterministic.
```

Comment:
Is row order ever guaranteed to be deterministic? So not sure what is meant here.  Should just say an id column will be added with column name [...]  no need to mention zipWithIndex.   Also, somewhere should explain what an id column is used for (e.g., to specify nearest neighbor vectors by associated id value).

Reply:
Revised the comment. Please check.